packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -413,6 +414,16 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                                     const age = Date.now() - (snapshotCachedEntry.cacheEntryTime ??
                                         (Date.now() - 30 * 24 * 60 * 60 * 1000));
 
+                                    // In order to decrease the number of times we have to execute a snapshot refresh,
+                                    // if this is the summarizer and we have a cache entry but it is past the defaultFirstSummaryCacheExpiryTimeout,
+                                    // force the network retrieval instead as there might be a more recent snapshot available.
+                                    if (this.hostPolicy.summarizerClient &&
+                                        age > this.defaultFirstSummaryCacheExpiryTimeout) {
+                                        // eslint-disable-next-line @typescript-eslint/dot-notation
+                                        props["cacheSummarizerExpired"] = 1;
+                                        return undefined;
+                                    }
```

Comment:
This looks good, now that I think about this code, since it has to be done here, we have a few different cases in which we need to expire/not use the cache. It would be awesome to move this logic into one place, as we are just expiring the cache. It may be helpful to future engineers if we conglomerate this code into one place